packages/database/test/query.test.ts

```diff
@@ -3766,6 +4192,90 @@ describe('Query Tests', () => {
     );
   });
 
+  it('Integer keys behave numerically with endBefore.', done => {
+    const ref = getRandomNode() as Reference;
+    ref.set(
+      {
+        1: true,
+        50: true,
+        550: true,
+        6: true,
+        600: true,
+        70: true,
+        8: true,
+        80: true
+      },
+      () => {
+        ref.endBefore(null, '50').once('value', s => {
+          expect(s.val()).to.deep.equal({
+            1: true,
+            6: true,
+            8: true
+          });
+          done();
+        });
+      }
+    );
+  });
+
+  it('Integer keys behave numerically with endBefore with underflow.', done => {
+    const ref = getRandomNode() as Reference;
+    ref.set(
+      {
+        1: true,
+        50: true,
+        550: true,
+        6: true,
+        600: true,
+        70: true,
+        8: true,
+        80: true
+      },
+      () => {
+        ref.endBefore(null, '' + INTEGER_32_MIN).once('value', s => {
+          expect(s.val()).to.deep.equal(null);
+          done();
+        });
+      }
+    );
+  });
+
+  it('Integer keys behave numerically with endBefore at boundary.', done => {
+    const ref = getRandomNode() as Reference;
+    ref.set(
+      {
+        1: true,
+        50: true,
+        550: true,
+        6: true,
+        600: true,
+        70: true,
+        8: true,
+        80: true,
+        'a': true
+      },
```

Comment:
It might make sense to extract a local variable here so you can use the same set twice.